### ISSUE TITLE
Greet users via direct message when possible

### ISSUE THREAD

**Issue by patcon** [publiclab/plotsbot#104]
Hi!

Just spending my first chunk of time of the public lab chat, and wondering whether there might be willingness to consider being a little less vocal with greetings?

(I'm a developer and a big fan of bots btw! I help maintain a few within organizations I work with :slightly_smiling_face:  )

Here are some :mag: facts, :heart: feelings, and :bulb: ideas floating in my head:

1. :mag: plotsbot engages through IRC.
1. :mag: plotsbot can DM IRC users
1. :mag: plotsbot cannot currently DM users of other platforms.
1. :mag: Gitter has a feature to present a descriptive pop-up message to users on very first join.
1. :heart: I feel that some bot actions are strategic to do publicly (to be performative), even though private interaction might be possible.
1. :heart: I feel that greeting of users would be just as effective if done privately.
1. :heart: I feel that looking backward through history for interesting convo is difficult with all the greetings.
1. :bulb: Allow plotsbot to greet via DM on IRC when possible, and greet publicly for other platforms.
1. :bulb: In :crystal_ball: future, consider extending ability to greet privately to other platforms.
1. :bulb: Disable greeting of Gitter users, and use the descriptive popup instead.

Anyhow, curious your thoughts! Thanks so much for all the work creating this bot -- I know it does a whole lot more that I'm just learning about. :)

**Comment by ryzokuken** [comment]
Hi, @patcon!

This looks great, I personally really appreciate the time and effort you've put into this.
Looking at the logs, one could not refute your observation. Let's discuss (either synchronously on chat or asynchronously in here) about your ideas, I would love to talk this one out and hopefully, see you getting involved with the project (we could definitely use an extra pair of hands around).

**Comment by pdurbin** [comment]
I'm also just now spending time in the chat room for the first time and I feel like plotbot is pretty noisy, especially the "Hi, looks like nobody's around just now" messages. I'm way less bothered by the welcome message when new people join, even though it isn't a direct message, which is what this issue seems to be about mostly.

**Comment by patcon** [comment]
Good call. Should I migrate this issue to be more generally about being able to DM when possible? I don't feel it needs to be specific to greetings :)

**Comment by patcon** [comment]
And @ryzokuken, I'll likely come back to help with this once I'm spending more time in chat. I'm not actively working on PL things with now, but imagine I'll eventually overlap due to my interests in community animating. At that point, coexisting with the bot will be on my critical path, and I'll be excited to help!

**Comment by ryzokuken** [comment]
@patcon I'd love that! Thanks.

**Comment by pdurbin** [comment]
@patcon I love your writeup but since I have the sense that you're focused on a solution (direct messages) rather than a problem (plotsbot being too vocal), I opened a new issue about what I consider the problem to be (too many "looks like nobody's around just now" messages) in #108. I'd love to hear what you and @ryzokuken and others think.

**Comment by pdurbin** [comment]
@ryzokuken and I have been talking about how it's challenging to switch to direct messages given the current chat services. See also "Where is the home base for chat? IRC, Gitter, Matrix, Slack, etc.?" at https://github.com/publiclab/plots2/issues/2590
